Milestone4_Clean Code/clean_code.md

```diff
@@ -0,0 +1,134 @@
+2. It may take developers more time to understand the code, and it will be difficult to find the issues.
+3. It affects maintaining the code, since the variable names are poorly defined, and introducing new functionality will require first understanding the original code.
+
+**How did refactoring improve code readability?**
+1. It explained the purpose of the variables used and the function written.
+2. It was easier to read the code and modify it.
+3. The complexity of the code was decreased, and it was easy to analyze the code step by step.
+
+
+# Writing Small, Focused Functions
+
+**Why is breaking down functions beneficial?**
+1. It makes the code easy to understand, like which block of code is for what purpose.
+2. Breaking down a big function into small ones makes it reusable.
+3. It is easy to debug, find a bug, and resolve the issue.
+4. Also, it maintains the code for future use. The code can be easily modified based on new functionality without affecting the old one.
+
+**How did refactoring improve the structure of the code?**
+1. Code duplication was avoided by making a new function for the block of code that was used frequently. Thus, this function can be reused.
+2. All the small functions had their specific functionality to perform.
+
+
+# Avoiding Code Duplication
+
+**What were the issues with duplicated code?**
+1. Unnecessarily, the same code was written repeatedly, which reduced readability.
+2. Since the same code was written in multiple places, if there is a code fix for that block of code, it has to be fixed in multiple places.
+3. It increased effort to understand and also complexity for the main function.
+
+**How did refactoring improve maintainability?**
+1. The repeated code of the block was created into a function, thus making it reusable.
+2. It was easy to understand the code and increased readability.
+3. Fixing the code was easier as it had to be modified in one place.
+
+
+# Refactoring Code for Simplicity
+
+**What made the original code complex?**
+1. The original code consisted of unnecessary if-else conditions.
+2. The code was difficult to understand and took more effort to read.
+3. The code had duplicated code within its if-else conditions, making it difficult to maintain.
+
+**How did refactoring improve it?**
+1. If-else conditions were removed and replaced with return statements, which made it easier to understand.
+2. It increased maintainability, and it became easier to add new functionality into it.
+3. It looked more structured and organized.
+
+
+# Commenting & Documentation
+
+**When should you add comments?**
+1. When the task of the code is not clear by the code itself.
+2. When a code is added specific to an issue, then why and what code changes were made to resolve the issue.
+3. When a new code replaces old code, it has to be specified why the code was modified.
+4. Sometimes a 'TODO' comment is added just to keep track that some changes have to be done at this specific location.
```

Comment:
make sure the TODO references a github issue otherwise the work will not get prioritised